### PR TITLE
[Package] Use cpack to build deb package

### PR DESCRIPTION
**Pull request by ityuhui**
Use cmake/cpack to build debian package. Thanks @ahmedalnuaimi @ahmedyarub for this suggestion !

After these files are merged, we can use the command:
`cpack -G DEB`
to generate a debian package.

Mark this PR as `WIP` because I will commit the files `kubernetes/CMakeLists.txt` and `kubernetes/Packing.cmake` to openapi-generator project.

### CODE REVIEW

**Comment by ityuhui** [comment]
cc @ahmedalnuaimi @ahmedyarub 

Could you please help review this PR too ? Thank you.

**Comment by ityuhui** [comment]
```
root@k8s-node1:~/c/kubernetes/build# dpkg-deb -c ./libkubernetes_0.0.1_amd64.deb
drwxr-xr-x root/root         0 2021-11-18 16:49 ./usr/
drwxr-xr-x root/root         0 2021-11-18 16:49 ./usr/local/
drwxr-xr-x root/root         0 2021-11-18 16:49 ./usr/local/include/
drwxr-xr-x root/root         0 2021-11-18 16:49 ./usr/local/include/kubernetes/
drwxr-xr-x root/root         0 2021-11-18 16:49 ./usr/local/include/kubernetes/api/
-rw-r--r-- root/root       351 2021-11-05 13:57 ./usr/local/include/kubernetes/api/AdmissionregistrationAPI.h
-rw-r--r-- root/root      5156 2021-11-05 13:57 ./usr/local/include/kubernetes/api/AdmissionregistrationV1API.h
...
drwxr-xr-x root/root         0 2021-11-18 16:49 ./usr/local/include/kubernetes/watch/
-rw-r--r-- root/root       334 2021-07-26 16:51 ./usr/local/include/kubernetes/watch/watch_util.h
drwxr-xr-x root/root         0 2021-11-18 16:49 ./usr/local/include/kubernetes/websocket/
-rw-r--r-- root/root       345 2021-09-14 20:35 ./usr/local/include/kubernetes/websocket/kube_exec.h
-rw-r--r-- root/root      1017 2021-09-13 14:46 ./usr/local/include/kubernetes/websocket/wsclient.h
drwxr-xr-x root/root         0 2021-11-18 16:49 ./usr/local/lib/
drwxr-xr-x root/root         0 2021-11-18 16:49 ./usr/local/lib/cmake/
drwxr-xr-x root/root         0 2021-11-18 16:49 ./usr/local/lib/cmake/kubernetes/
-rw-r--r-- root/root       156 2021-11-10 16:07 ./usr/local/lib/cmake/kubernetes/kubernetesConfig.cmake
-rw-r--r-- root/root      1269 2021-11-10 16:07 ./usr/local/lib/cmake/kubernetes/kubernetesConfigVersion.cmake
-rw-r--r-- root/root       998 2021-11-18 16:49 ./usr/local/lib/cmake/kubernetes/kubernetesTargets-noconfig.cmake
-rw-r--r-- root/root      3287 2021-11-10 16:07 ./usr/local/lib/cmake/kubernetes/kubernetesTargets.cmake
-rw-r--r-- root/root   3010216 2021-11-10 16:08 ./usr/local/lib/libkubernetes.so
```

**Comment by ityuhui** [comment]
```
root@k8s-node1:~/c/kubernetes/build# dpkg-deb -I ./libkubernetes_0.0.1_amd64.deb  
 new Debian package, version 2.0.
 size 867058 bytes: control archive=17539 bytes.
     240 bytes,     9 lines      control              
   57433 bytes,   610 lines      md5sums              
 Package: libkubernetes
 Version: 0.0.1
 Section: devel
 Priority: optional
 Architecture: amd64
 Installed-Size: 3876
 Maintainer: https://github.com/kubernetes-client/c
 Description: The Kubernetes client library for the C programming language.
```

File: kubernetes/Packing.cmake [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+set(CPACK_PACKAGE_VERSION_MAJOR ${PROJECT_VERSION_MAJOR})
+set(CPACK_PACKAGE_VERSION_MINOR ${PROJECT_VERSION_MINOR})
+set(CPACK_PACKAGE_VERSION_PATCH ${PROJECT_VERSION_PATCH})
+
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY ${PROJECT_PACKAGE_DESCRIPTION_SUMMARY})
+set(CPACK_PACKAGE_VENDOR ${PROJECT_PACKAGE_VENDOR})
+set(CPACK_PACKAGE_CONTACT ${PROJECT_PACKAGE_CONTACT})
+set(CPACK_DEBIAN_PACKAGE_MAINTAINER ${PROJECT_PACKAGE_MAINTAINER})
+
+set(CPACK_VERBATIM_VARIABLES YES)
```
Comment (ahmedyarub) [inline kubernetes/Packing.cmake:12]: Enable adding required dependencies' info using the following variable:
`set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS TRUE)`
Note that on my machine this only added OpenSSL. You might consider adding libyaml and libwebsockets manually using `CPACK_DEBIAN_PACKAGE_DEPENDS`.

**Review by ahmedyarub — COMMENTED** [review]
Tested it and everything worked as expected. We might need to consider adding dependency info though.

File: kubernetes/Packing.cmake (hunk 1 above)
Comment (ityuhui) [reply kubernetes/Packing.cmake:12]: Thank you @ahmedyarub 

`set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS TRUE)` is good, in my build env, it generates the depenencies:
```
Depends: libc6 (>= 2.17), libcurl4 (>= 7.56.1), libssl1.1 (>= 1.1.0)
```

Currently (before Ubuntu 22.04 LTS releases), `libwebsockets` must be installed from source code for the C client. So `CPACK_DEBIAN_PACKAGE_DEPENDS` is not needed now.

**Comment by ityuhui** [comment]
```
root@k8s-node1:~/c/kubernetes/build# dpkg-deb -I ./libkubernetes_0.0.1_amd64.deb 
 new Debian package, version 2.0.
 size 867100 bytes: control archive=17596 bytes.
     309 bytes,    10 lines      control              
   57433 bytes,   610 lines      md5sums              
 Package: libkubernetes
 Version: 0.0.1
 Section: devel
 Priority: optional
 Architecture: amd64
 Depends: libc6 (>= 2.17), libcurl4 (>= 7.56.1), libssl1.1 (>= 1.1.0)
 Installed-Size: 3876
 Maintainer: https://github.com/kubernetes-client/c
 Description: The Kubernetes client library for the C programming language.
```

**Comment by brendandburns** [comment]
/lgtm
/approve

This is great!

**Comment by ityuhui** [comment]
The files `kubernetes/CMakeLists.txt` and `kubernetes/Packing.cmake` have been committed to openapi-generator by https://github.com/OpenAPITools/openapi-generator/pull/10935

The regenerated files (`kubernetes/CMakeLists.txt` and `kubernetes/Packing.cmake`) are same with the files in this PR exactly. So there is no need to modify any file of this PR.

Hi @brendandburns 
This PR is ready to merge now.

**Comment by brendandburns** [comment]
/lgtm
/approve

Thanks!

**Comment by ityuhui** [comment]
Removed the label `do-not-merge/hold` to trigger the merging because @brendandburns has approved the latest code.

**Pull request merged**